src/store/timetravel.js

```diff
@@ -0,0 +1,118 @@
+const serializeState = (state) => {
+  const scaleX = state.application.scale.x;
+  const scaleY = state.application.scale.y;
+
+  const clone = JSON.parse(JSON.stringify(state));
+  clone.application.scale.x = scaleX;
+  clone.application.scale.y = scaleY;
```

Comment:
Is this section just about making a deep copy? Seems like that should be possible without saving off and replacing the scales. Does `const clone = Object.assign({}, state)` not do what you want?